ts/features/lollipop/utils/fetch.ts

```diff
@@ -0,0 +1,288 @@
+import URLParse from "url-parse";
+import { PublicKey, sign } from "@pagopa/io-react-native-crypto";
+import {
+  LollipopConfig,
+  chainSignPromises,
+  SignPromiseResult,
+  getSignAlgorithm,
+  toSignatureComponents
+} from "..";
+import { KeyInfo } from "../../../utils/crypto";
+import { toFetchTimeout, toRetriableFetch } from "../../../utils/fetch";
+import { generateDigestHeader } from "../../../utils/httpSignature/digest";
+import {
+  generateSignatureBase,
+  SignatureBaseResult
+} from "../../../utils/httpSignature/signature";
+import { SignatureConfig } from "../../../utils/httpSignature/types/SignatureConfig";
+
+/**
+ * Decorates the current fetch with LolliPOP headers and http-signature
+ */
+export const lollipopFetch = (
```

Comment:
I suggest to add a TODO with a linked jira task related to track a refactor in a more functional way of this function